src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogEntryWrittenTest.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using Xunit;
+
+namespace System.Diagnostics.Tests
+{
+    public class EventLogEntryEventWrittenTest
+    {
+        static AutoResetEvent signal;
+        private const string message = "EventLogEntryEventWrittenTestMessage";
+        private int eventCounter;
+
+        public void RaisingEvent(string log, string methodName, bool waitOnEvent = true)
+        {
+            signal = new AutoResetEvent(false);
+            eventCounter = 0;
+            string source = "Source_" + methodName;
+
+            try
+            {
+                EventLog.CreateEventSource(source, log);
+                using (EventLog eventLog = new EventLog())
+                {
+                    eventLog.Source = source;
+                    eventLog.EntryWritten += new EntryWrittenEventHandler((object sourceObject, EntryWrittenEventArgs e) =>
+                    {
+                        eventCounter += 1;
+                        signal.Set();
+                    });
+                    eventLog.EnableRaisingEvents = waitOnEvent;
+                    eventLog.WriteEntry(message, EventLogEntryType.Information);
+                    if (waitOnEvent)
+                    {
+                        if (!signal.WaitOne(6000))
```

Comment:
The delays need a comment, as it is not obvious. Also a link to the doc explaining the delay.